onnx/backend/test/cpp/driver/test_driver.cc

```diff
@@ -0,0 +1,180 @@
+#include <fstream>
+#include "test_driver.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <dirent.h>
+#include <errno.h>
+#endif
+
+#define ERRNO_DIR_END 0
+
+bool fileExist(const std::string& filename){
+	FILE *fp;
+	if ((fp = fopen(filename.c_str(), "r")) != NULL){
+		fclose(fp);
+		return true;
+	}
+	return false;
+}
+
+void TestDriver::setDefaultDir(const std::string &s){
+	default_dir_ = s;
+}
+
+//load single test case in case_dir to _testcases
+int TestDriver::fetchSingleTestCase(const std::string& case_dir) {
+  std::string model_name = case_dir;
+  model_name += "model.onnx";
+  if (fileExist(model_name)) {
+    TestCase test_case;
+    test_case.model_filename_ = model_name;
+    test_case.model_dirname_ = case_dir;
+    int case_count = 0;
+    for (;; case_count++) {
+      std::vector<std::string> input_filenames, output_filenames;
+      std::string input_name, output_name;
+      std::string case_dirname = case_dir;
+      case_dirname += "test_data_set_" + std::to_string(case_count);
+
+      output_name = case_dirname + "/output_" + "0" + ".pb";
+      if (!fileExist(output_name)) {
+        break;
+      }
+
+      for (int data_count = 0;; data_count++) {
+        input_name =
+            case_dirname + "/input_" + std::to_string(data_count) + ".pb";
+        output_name =
+            case_dirname + "/output_" + std::to_string(data_count) + ".pb";
+        if (!fileExist(input_name) && !fileExist(input_name)) {
+          break;
+        }
+        if (fileExist(input_name)) {
+          input_filenames.push_back(input_name);
+        }
+        if (fileExist(output_name)) {
+          output_filenames.push_back(output_name);
```

Comment:
What if we have input_file but not output_file?

Reply:
I guess it doesnt make sense if a node has no output? But I will check this at the beginning of this chunk of code.